### DISCUSSION TITLE
How to match a request with query param

### DISCUSSION THREAD

**Discussion by sydneyhenrard** [mockoon/mockoon#1367] [Help] [answered]
Hello,

I try to configure a mock for a request like `GET /test?x=1`. I configure the mock as:
![image](https://github.com/mockoon/mockoon/assets/5309640/bdf68e2d-12b9-45ec-97a2-523c5d875a14)

But when I call http://localhost:3001/test?x=2 I get the mock response which I don't expect. I expect to get a 404 since x is not equal to 1.

Thanks
Sydney

**Comment by 255kb** [comment, marked as answer]
Hi, If you have a single route response, it will always get served. 
You must have at least two responses, a default one (blue flag), and some other with rules. 
I suggest that you have a look at our [reponse rules tutorial](https://mockoon.com/tutorials/responses-and-rules/) to see a typical setup.  
I admit that having rules on a single response can be confusing.

**Comment by sydneyhenrard** [comment]
I managed to make it work by adding a 404 response by default. I am evaluating WireMock vs Mockoon as they are very similar for my use cases. The requirement of having a default 404 rule is a real constraint. If the request does not match, it's expected to get a 404 by default as it is in WireMock. Anyway Mockoon is a good alternative. Thx for your quick answer

**Reply by 255kb** [reply]
You could eventually activate the [fallback mode](https://mockoon.com/docs/latest/route-responses/multiple-responses/#fallback-mode) on the route (keeping 1 response with a rule) and create a catch-all with a 404. 

You will find an exemple setup that you can directly open in Mockoon (route order matters):
[demo-default-404.json](https://github.com/mockoon/mockoon/files/15110907/demo-default-404.json)
